provider.go

```diff
@@ -5,110 +5,179 @@ package azuresecrets
 
 import (
 	"context"
+	"fmt"
+	"net/http"
 	"time"
 
-	"github.com/Azure/azure-sdk-for-go/services/preview/authorization/mgmt/2018-09-01-preview/authorization"
-	"github.com/Azure/go-autorest/autorest"
-	"github.com/Azure/go-autorest/autorest/azure/auth"
-	"github.com/hashicorp/vault-plugin-secrets-azure/api"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/vault/sdk/logical"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization"
```

Comment:
I believe we're going to need to upgrade this to `github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v2` per guidance in https://github.com/Azure/azure-sdk-for-go/issues/18674. The reason is that when running the acceptance tests you'll receive an error like `"UnsupportedApiVersionForRoleDefinitionHasDataActions"` when assigning roles with dataActions/notDataActions. This is a behavior we'll want to maintain (had a customer issue when we accidentally removed it in the past, hence the test coverage for it 😅).

It appears there is some breakage (mostly struct naming?) in the v2 package that'll need some adjustments. Lmk if you need any help or if I can explain better!

Reply:
Updated!